packages/autocomplete-plugin-algolia-insights/src/types/InsightsClient.ts

```diff
@@ -1 +1,30 @@
-export type InsightsClient = any;
+import type {
+  InsightsMethodMap,
+  InsightsClient as _InsightsClient,
+} from 'search-insights';
+
+export type {
+  Init as InsightsInit,
+  AddAlgoliaAgent as InsightsAddAlgoliaAgent,
+  SetUserToken as InsightsSetUserToken,
+  GetUserToken as InsightsGetUserToken,
+  OnUserTokenChange as InsightsOnUserTokenChange,
+} from 'search-insights';
+
+export type InsightsClientMethod = keyof InsightsMethodMap;
+
+export type InsightsClientPayload = {
+  eventName: string;
+  queryID: string;
+  index: string;
+  objectIDs: string[];
+  positions?: number[];
+};
+
+type QueueItemMap = Record<string, unknown>;
```

Comment:
This could be more strictly typed, but it requires updating TypeScript, which then triggers other type issues.

Let's handle this in another PR.